### PR TITLE
feat(chats): wire up UI for replies, typing indicator, and attachments

### PR DESCRIPTION
**Pull request by sdwoodbury**
### What this PR does 📖

- typing indicator: shows in chatbar and sidebar. displays the user's icon rather than the username. multiple typing indicators, such as for group messages, have not yet been implemented. 
- replies: see pictures 
- attachments: allow the user to attach and send multiple files. prior to sending, attachments may be removed via the trashcan icon. 

### Which issue(s) this PR fixes 🔨

- Resolve #

### Additional comments 🎤

image
image
image

### CODE REVIEW

**Comment by InfamousVague** [comment]
there are a few small things left to wire but I think we can safely merge this in when you're ready @sdwoodbury

File: common/src/warp_runner/manager/commands/raygun_commands.rs
Comment (dariusc93) [inline common/src/warp_runner/manager/commands/raygun_commands.rs:140]: It might be better for `DownloadAttachment` to use a single `PathBuf` representing the location and file where it would be stored at. I do see the reason for it and its not blocking to the PR, but just generally a suggestion.

**Review by dariusc93 — DISMISSED** [review]
Looked it over and overall looks good.

File: common/src/warp_runner/manager/commands/raygun_commands.rs
Comment (dariusc93) [inline common/src/warp_runner/manager/commands/raygun_commands.rs:128]: Just as a quick note, since I just noticed it, messages here in the attach function are optional. So down the road, might want to lift it up so it could send an attachment even if a message is not present for `Raygun::attach`. Doesnt block this PR though unless its something you wish to quickly do.

File: common/src/warp_runner/manager/commands/raygun_commands.rs
Comment (sdwoodbury) [reply common/src/warp_runner/manager/commands/raygun_commands.rs:140]: Agreed. Especially if the user wants to rename the file when downloading it.

**Comment by phillsatellite** [comment]
Just a couple things I found when testing, since this is adding in a lot as it is if it'd be easier to fix this in a new PR just lemme know and I will write up a ticket. Other than these small issues this PR is super cool!! 🎉 

(These are on Windows and MacOS)
1. When the user adds attachments they can't actually send the message unless they type some text into the input box

https://user-images.githubusercontent.com/93608357/221927355-eed33484-4ea0-49ec-8f75-f04c08a8fb1f.mov

2. When someone is on a smaller monitor or laptop, the message you type up with the attachments can look a bit messy but I do not think this is block worthy of the PR 

Screenshot 2023-02-28 at 12 12 23 PM

**Review by dariusc93 — APPROVED** [review]

**Review by phillsatellite — APPROVED** [review]

**Comment by phillsatellite** [comment]
Awesome fix! 🔥

**Pull request merged**
